core/ir/x86/instr.c

```diff
@@ -559,7 +560,7 @@ bool
 instr_is_return(instr_t *instr)
 {
     int opc = instr_get_opcode(instr);
-    return (opc == OP_ret || opc == OP_ret_far || opc == OP_iret);
+    return (opc == OP_ret || opc == OP_ret_far || opc == OP_sysret || opc == OP_iret);
```

Comment:
+ What about sysexit?
+ If sysret is a return: would we then expect syscall to be a call?  And sysenter?

Neither sysret nor sysexit takes the address from the stack nor presumably involves hardware call-return prediction: it may be better to have both of these be generic indirect branches.

If we wanted symmetric user-kernel handling we'd also make syscall/sysenter be indirect branches, and I guess int and into and int3??  But they all seem different from a user-mode-centric viewpoint so I guess we leave them alone and just mark sysret/sysexit?

Reply:
Re sysexit: Though I didn't find it in the current QEMU traces, it makes sense to also treat it the same.

Re syscall/sysenter: I agree. Added comments.

Reply:
> Re syscall/sysenter: I agree. Added comments.

I mean I agree with your latter though. No need to treat syscall/sysenter as indirect branches or calls.